src/model/sponsored_message.rs

```diff
@@ -67,10 +67,7 @@ impl SponsoredMessage {
 
         // TODO: Support multiple sponsored messages
         if let Some(sponsored_message) = sponsored_messages.messages.first() {
-            Ok(Some(Self::new(
-                sponsored_message,
-                &session.chat(sponsored_message.sponsor_chat_id),
-            )))
+            Ok(Some(Self::new(sponsored_message, &session.chat(0))))
```

Comment:
Is the sponsored chat id always 0?

Reply:
This was because there is now four different sponsor types (https://github.com/paper-plane-developers/tdlib-rs/blob/bba2c1c992f648caa6f91ed8ad0c23b206c4670e/tdlib/tl/api.tl#L1261-L1271), and just one of them (`messageSponsorTypePublicChannel`) has the legacy chat_id property that was previously used.
I just put a `0` there because we would need to discuss the best approach in this case.